scripts/galsim_baker.py

```diff
@@ -0,0 +1,62 @@
+"""
+Use GalSim to compute colors and execution times for LSST bands for a
+grid of redshifts and the provided library of SEDs.
+"""
+import os
+from collections import OrderedDict
+import time
+import numpy as np
+import galsim
+
+# Numbers for LSST.
+_effective_diameter = 667. # cm
+_exptime = 15. # s
+
+class GalSimPhotometry(object):
+    """
+    Class to compute colors for LSST bands using the GalSim
+    chromaticity tools.
+    """
+    def __init__(self, photometry_baker_dir='..'):
+        """
+        Read in the throughput data and SED filenames from the package
+        data subfolder.
+        """
+        # Read throughput data for each band.
+        bp_dir = os.path.join(photometry_baker_dir, 'data', 'bandpass')
+        self.bandpasses = OrderedDict()
+        for band in 'ugrizy':
+            bp = galsim.Bandpass(os.path.join(bp_dir, 'total_%s.dat' % band))
+            self.bandpasses[band] = bp.withZeropoint('AB', _effective_diameter,
+                                                     _exptime)
+
+        # Read the SED filenames for the wavelengths in nm.
+        self.sed_dir = os.path.join(photometry_baker_dir, 'data', 'sed')
+        self.sed_names = sorted([x for x in os.listdir(self.sed_dir)
+                                 if 'spec' in x and 'ang' not in x])
+
+    def colors(self, sed_name, redshift):
+        """
+        Return the ug, gr, ri, iz, zy colors and execution times for
+        the specifed SED and redshift.
+        """
+        sed_fn = os.path.join(self.sed_dir, sed_name)
+        sed = galsim.SED(sed_fn).atRedshift(redshift)
```

Comment:
I have to admit, we never really looked into the speed of any of this code.  It's always extremely subdominant to anything else we are doing in terms of rendering images.  But I would guess that switching to linear rather than spline would be faster.  There may also be dumb inefficiencies in various places that we never noticed.